x-pack/plugins/session_view/public/components/SessionView/hooks.ts

```diff
@@ -49,19 +42,15 @@ export const useParseSessionViewProcessEvents = (getData: ProcessEventResults |
   };
 
   useEffect(() => {
-    if (!getData) {
-      return;
-    }
-
-    const events: ProcessEvent[] = (getData.events?.hits || []).map(
+    const events: ProcessEvent[] = (getData?.events?.hits || []).map(
       (event: any) => event._source as ProcessEvent
     );
-    const alerts: ProcessEvent[] = (getData.alerts?.hits || []).map((event: any) => {
+    const alerts: ProcessEvent[] = (getData?.alerts?.hits || []).map((event: any) => {
       return event._source as ProcessEvent;
     });
     const all: ProcessEvent[] = events.concat(alerts).sort(sortEvents);
     setData(all);
-  }, [getData]);
+  }, [getData?.events, getData?.alerts]);
```

Comment:
can we just just have `getData` as the dependency?

Reply:
it was causing infinite re-renders because I also added `sessionLeader` into getData when fetching session view events. i will be reverting that part (fetching sessionLeader) and this change here since it was mentioned above by Karl that it won't always be available to us.

Reply:
I see in that case it might be better to do something like
```
const { events, alerts } = getData || {};
...
useEffect(() => {
  const events = events?.hits.map(() => {});
  ...
}, [events, alerts]);
```

so we're not calling `getData?.events` everywhere

Reply:
You are right, this is better 👍